raiden/transfer/channel.py

```diff
@@ -914,32 +918,58 @@ def is_valid_unlock(
     return result
 
 
-def is_valid_action_withdraw(
-    channel_state: NettingChannelState, withdraw: ActionChannelWithdraw
+def is_valid_total_withdraw(
+    channel_state: NettingChannelState, our_total_withdraw: WithdrawAmount
 ) -> SuccessOrError:
     balance = get_balance(sender=channel_state.our_state, receiver=channel_state.partner_state)
 
     withdraw_overflow = not is_valid_channel_total_withdraw(
-        TokenAmount(withdraw.total_withdraw + channel_state.partner_total_withdraw)
+        TokenAmount(our_total_withdraw + channel_state.partner_total_withdraw)
     )
 
-    withdraw_amount = withdraw.total_withdraw - channel_state.our_total_withdraw
+    withdraw_amount = our_total_withdraw - channel_state.our_total_withdraw
     if get_status(channel_state) != ChannelState.STATE_OPENED:
         return SuccessOrError("Invalid withdraw, the channel is not opened")
     elif withdraw_amount <= 0:
-        return SuccessOrError(f"Total withdraw {withdraw.total_withdraw} did not increase")
+        return SuccessOrError(f"Total withdraw {our_total_withdraw} did not increase")
     elif balance < withdraw_amount:
         return SuccessOrError(
             f"Insufficient balance: {balance}. Requested {withdraw_amount} for withdraw"
         )
     elif withdraw_overflow:
         return SuccessOrError(
-            f"The new total_withdraw {withdraw.total_withdraw} will cause an overflow"
+            f"The new total_withdraw {our_total_withdraw} will cause an overflow"
         )
     else:
         return SuccessOrError()
 
 
+def is_valid_action_withdraw(
+    channel_state: NettingChannelState, withdraw: ActionChannelWithdraw
+) -> SuccessOrError:
+    return is_valid_total_withdraw(channel_state, withdraw.total_withdraw)
+
+
+def is_valid_action_coopsettle(
+    channel_state: NettingChannelState,
+    coop_settle: ActionChannelCoopSettle,  # pylint: disable=unused-argument
+    total_withdraw: WithdrawAmount,
+) -> SuccessOrError:
+    result = is_valid_total_withdraw(channel_state, total_withdraw)
```

Comment:
This will fail if we don't have anything to withdraw in our channel. It is not very useful to settle a channel if you have no balance left, but it might still be nice to allow it.

It is also not obvious why we check the total_withdraw for one participant but not the other.

Reply:
To me it looks like `is_valid_total_withdraw` checks both sides.